File: core/src/main/java/com/kin/ecosystem/core/data/settings/SettingsDataSourceImpl.java
package com.kin.ecosystem.core.data.settings;

import android.support.annotation.NonNull;

public class SettingsDataSourceImpl implements SettingsDataSource {

	private final SettingsDataSource.Local local;

	public SettingsDataSourceImpl(@NonNull final SettingsDataSource.Local local) {
		this.local = local;
	}

	@Override
	public void setIsBackedUp(final String publicAddress, boolean isBackedUp) {
		local.setIsBackedUp(publicAddress, isBackedUp);
	}

	@Override
	public boolean isBackedUp(final String publicAddress) {
		return local.isBackedUp(publicAddress);
	}

}

File: core/src/main/java/com/kin/ecosystem/core/data/settings/SettingsDataSourceImpl.kt
package com.kin.ecosystem.core.data.settings

class SettingsDataSourceImpl(private val local: SettingsDataSource.Local) : SettingsDataSource {

    override fun setIsBackedUp(publicAddress: String, isBackedUp: Boolean) {
        local.setIsBackedUp(publicAddress, isBackedUp)
    }

    override fun isBackedUp(publicAddress: String): Boolean {
        return local.isBackedUp(publicAddress)
    }

    override fun setSawOnboarding(kinUserId: String) {
        local.setSawOnboarding(kinUserId)
    }

    override fun isSawOnboarding(kinUserId: String): Boolean {
        return local.isSawOnboarding(kinUserId)
    }
}
